http/src/test/scala/org/broadinstitute/dsde/workbench/leonardo/service/AuthProviderSpec.scala

```diff
@@ -107,7 +107,7 @@ class AuthProviderSpec extends FreeSpec with ScalatestRouteTest with Matchers wi
       leo.deleteCluster(userInfo, project, cluster1Name).futureValue
 
       //verify we correctly notified the auth provider
-      verify(spyProvider).notifyClusterCreated(any[ClusterInternalId], any[WorkbenchEmail], any[GoogleProject], any[ClusterName])(any[ExecutionContext])
+      verify(spyProvider).notifyClusterCreated(any[ClusterInternalId], any[WorkbenchEmail], any[GoogleProject], any[ClusterName], any[TraceId])(any[ExecutionContext])
```

Comment:
The line above is where `NullPointerException` is coming from.

Reply:
I have a theory about what's going on. The key is in the error message.
```
[info] This message may appear after an NullPointerException if the last matcher is returning an object 
[info] like any() but the stubbed method signature expect a primitive argument, in this case,
[info] use primitive alternatives.
[info]     when(mock.get(any())); // bad use, will raise NPE
[info]     when(mock.get(anyInt())); // correct usage use
```

In fact, the last argument is `any[TraceId]`, an object. `TraceId` is a Scala value object, meaning at runtime it is a primitive. So this message applies -- the method expects a primitive but our matcher returns an object.

This seems to be an example where Mockito (a Java library) doesn't play nice with Scala language  features (values objects in this case).

I tried a local change where I made my own `TraceId` in `LeonardoModel.scala` as:
```
case class TraceId(uuid: UUID)
```
E.g. it's _not_ a value object. When I use this class, mockito is happy and the test passes. I'm happy to PR that change into this branch.

Reply:
Made a PR, feel free to merge or not.

Reply:
ah...I would never thought this is the issue since the code executes just fine even though matcher actually failed.

It seems like `TraceId(any[UUID])` would work fine as matcher, so I think we should use this instead of creating a separate `TraceId`

Reply:
^ I didn't try that, it's worth a try. Agreed that would be preferable if it works.